### ISSUE TITLE
False positive for CA1508

### ISSUE THREAD

**Issue by gewarren** [dotnet/roslyn-analyzers#4408]
---
**Issue moved from dotnet/docs#21331**
- Please respond to @maxkoshevoi.

---

_From @maxkoshevoi on Tuesday, November 3, 2020 9:24:30 PM_

```cs
if (lessonInfo.Settings.Hiding.EventTypesToHide.Count == 0 &&
    lessonInfo.Settings.Hiding.TeachersToHide.Count == 0)
{
    return true;
}
```
Where `TeachersToHide` and `EventTypesToHide` are both `List<long>`

![image](https://user-images.githubusercontent.com/6609929/98042060-90669100-1e2b-11eb-93c9-af4613bb9313.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 44bae70d-365f-c01d-acae-7d7ae9c0ef03
* Version Independent ID: 1fd1a3b5-e990-2644-f12c-2f4a06e053c3
* Content: [CA1508: Avoid dead conditional code (code analysis) - .NET](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508)
* Content Source: [docs/fundamentals/code-analysis/quality-rules/ca1508.md](https://github.com/dotnet/docs/blob/master/docs/fundamentals/code-analysis/quality-rules/ca1508.md)
* Product: **dotnet**
* GitHub Login: @mavasani
* Microsoft Alias: **mavasani**

**Comment by gewarren** [comment]
---
**Issue moved from dotnet/docs#21331**
- Please respond to @Youssef1313.

---

_From @Youssef1313 on Wednesday, November 4, 2020 10:37:47 AM_

This needs to be moved to dotnet/roslyn-analyzers repository.

**Comment by mavasani** [comment]
This needs a much more clear repro. @maxkoshevoi can you please provide a standalone repro that shows the false positive against 3.3.1 NuGet package?

**Comment by maxkoshevoi** [comment]
Here is a line of code where false positive occurs: https://github.com/maxkoshevoi/NureTimetable/blob/master/NureTimetable/NureTimetable/UI/ViewModels/Lessons/LessonSettings/LessonSettingsViewModel.cs#L141

This project hasFxCopAnalyzers 3.3.1 installed.

**Comment by maxkoshevoi** [comment]
@mavasani are you able to reproduce the issue with provided repository?

**Comment by mavasani** [comment]
@maxkoshevoi I haven't been able to get to it yet. Will try sometime later this week to get to CA1508 issues.

**Comment by mavasani** [comment]
@maxkoshevoi I was able to reproduce the issue, thanks.

**Comment by mavasani** [comment]
This should be fixed in the .NET 8 SDK and latest 8.xx.xx MS.CA.NetAnalyzers NuGet packages.
